ProcessLib/BoundaryCondition/BoundaryConditionCollection.h

```diff
@@ -41,8 +44,10 @@ class BoundaryConditionCollection final
         NumLib::LocalToGlobalIndexMap const& dof_table,
         unsigned const integration_order);
 
+    void preTimestep(const double t);
+
 private:
-    std::vector<NumLib::IndexValueVector<GlobalIndexType>> _dirichlet_bcs;
+    mutable std::vector<NumLib::IndexValueVector<GlobalIndexType>> _dirichlet_bcs;
```

Comment:
In the long run, probably we should omit `mutable` as much as possible. This can be done by relaxing the constness of several methods. In the current case this might extend to the time integration code. So I'd like to leave it for later refactoring.